### ISSUE TITLE
Deploying a daemonset to the `default` namespace with an RKE2/K3s v1.25 cluster with PSACT `rancher-restricted` does not error out as expected

### ISSUE THREAD

**Issue by markusewalker** [rancher/dashboard#8502]
**Setup**
- Rancher version: `v2.7-head` https://github.com/rancher/rancher/commit/8835c43d5d3e1599695a5d9bba989afac5c93436
- Browser type & version: Chrome

**Describe the bug**

After provisioning an RKE2 v1.25 node provider cluster with PSACT set to `rancher-restricted`, I went to validate that creating workloads would fail. Attempting to create a test deployment with `nginx:latest` in the `default` namespace failed as expected:

![image](https://user-images.githubusercontent.com/52591819/225766985-b0b26c5c-aa09-4c68-9727-a35971838a1c.png)

I went to do the same for daemonsets, but I see the following:

![image](https://user-images.githubusercontent.com/52591819/225767050-cba28a8e-e0a5-40b5-8594-bcf525949c75.png)

Curious, I went to run `kubectl get <daemonset> -n default` to see if I would get an error in the event logs, which I did:

```
> kubectl describe daemonset.apps/test -n default
Name:           test
Selector:       workload.user.cattle.io/workloadselector=apps.daemonset-default-test
Node-Selector:  <none>
Labels:         workload.user.cattle.io/workloadselector=apps.daemonset-default-test
Annotations:    deprecated.daemonset.template.generation: 1
Desired Number of Nodes Scheduled: 1
Current Number of Nodes Scheduled: 0
Number of Nodes Scheduled with Up-to-date Pods: 0
Number of Nodes Scheduled with Available Pods: 0
Number of Nodes Misscheduled: 0
Pods Status:  0 Running / 0 Waiting / 0 Succeeded / 0 Failed
Pod Template:
  Labels:  workload.user.cattle.io/workloadselector=apps.daemonset-default-test
  Containers:
   container-0:
    Image:        nginx:latest
    Port:         <none>
    Host Port:    <none>
    Environment:  <none>
    Mounts:       <none>
  Volumes:        <none>
Events:
  Type     Reason        Age                    From                  Message
  ----     ------        ----                   ----                  -------
  Warning  FailedCreate  5m23s                  daemonset-controller  Error creating: pods "test-hp4zz" is forbidden: violates PodSecurity "restricted:latest": unrestricted capabilities (container "container-0" must set securityContext.capabilities.drop=["ALL"]), runAsNonRoot != true (container "container-0" must not set securityContext.runAsNonRoot=false), seccompProfile (pod or container "container-0" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")
  Warning  FailedCreate  5m23s                  daemonset-controller  Error creating: pods "test-czk74" is forbidden: violates PodSecurity "restricted:latest": unrestricted capabilities (container "container-0" must set securityContext.capabilities.drop=["ALL"]), runAsNonRoot != true (container "container-0" must not set securityContext.runAsNonRoot=false), seccompProfile (pod or container "container-0" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")
  Warning  FailedCreate  5m23s                  daemonset-controller  Error creating: pods "test-lbktm" is forbidden: violates PodSecurity "restricted:latest": unrestricted capabilities (container "container-0" must set securityContext.capabilities.drop=["ALL"]), runAsNonRoot != true (container "container-0" must not set securityContext.runAsNonRoot=false), seccompProfile (pod or container "container-0" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")
  Warning  FailedCreate  5m23s                  daemonset-controller  Error creating: pods "test-lp8qt" is forbidden: violates PodSecurity "restricted:latest": unrestricted capabilities (container "container-0" must set securityContext.capabilities.drop=["ALL"]), runAsNonRoot != true (container "container-0" must not set securityContext.runAsNonRoot=false), seccompProfile (pod or container "container-0" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")
  Warning  FailedCreate  5m23s                  daemonset-controller  Error creating: pods "test-ncs2t" is forbidden: violates PodSecurity "restricted:latest": unrestricted capabilities (container "container-0" must set securityContext.capabilities.drop=["ALL"]), runAsNonRoot != true (container "container-0" must not set securityContext.runAsNonRoot=false), seccompProfile (pod or container "container-0" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")
  Warning  FailedCreate  5m23s                  daemonset-controller  Error creating: pods "test-dvt59" is forbidden: violates PodSecurity "restricted:latest": unrestricted capabilities (container "container-0" must set securityContext.capabilities.drop=["ALL"]), runAsNonRoot != true (container "container-0" must not set securityContext.runAsNonRoot=false), seccompProfile (pod or container "container-0" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")
  Warning  FailedCreate  5m23s                  daemonset-controller  Error creating: pods "test-2r5pn" is forbidden: violates PodSecurity "restricted:latest": unrestricted capabilities (container "container-0" must set securityContext.capabilities.drop=["ALL"]), runAsNonRoot != true (container "container-0" must not set securityContext.runAsNonRoot=false), seccompProfile (pod or container "container-0" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")
  Warning  FailedCreate  5m23s                  daemonset-controller  Error creating: pods "test-266s8" is forbidden: violates PodSecurity "restricted:latest": unrestricted capabilities (container "container-0" must set securityContext.capabilities.drop=["ALL"]), runAsNonRoot != true (container "container-0" must not set securityContext.runAsNonRoot=false), seccompProfile (pod or container "container-0" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")
  Warning  FailedCreate  5m22s                  daemonset-controller  Error creating: pods "test-xthkr" is forbidden: violates PodSecurity "restricted:latest": unrestricted capabilities (container "container-0" must set securityContext.capabilities.drop=["ALL"]), runAsNonRoot != true (container "container-0" must not set securityContext.runAsNonRoot=false), seccompProfile (pod or container "container-0" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")
  Warning  FailedCreate  2m39s (x7 over 5m21s)  daemonset-controller  (combined from similar events): Error creating: pods "test-lbtst" is forbidden: violates PodSecurity "restricted:latest": unrestricted capabilities (container "container-0" must set securityContext.capabilities.drop=["ALL"]), runAsNonRoot != true (container "container-0" must not set securityContext.runAsNonRoot=false), seccompProfile (pod or container "container-0" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")
```
As the logs indicate the expected failure, it is odd that the test daemonset does not reflect this error message outright, but only in the logs.

**To Reproduce**

1. As a standard user, provision a RKE2/K3s downstream node driver v1.25 cluster with PSACT set to `rancher-restricted`.
2. Validate that the cluster and nodes come up as `Active` in Cluster Management and Cluster Explorer.
3. Navigate to top-left menu > cluster > Workloads > Create.
4. Create a test daemonset with `nginx:latest` as the container image in the `default` namespace.
5. Note that it stays in `In Progress` state and does not error out as expected.
6. Click the `kubectl shell` and run command `kubectl describe <daemon set name> -n default`; validate the event logs show the expected error.

**Result**
The test daemonset is stuck in `In Progress` with no error message shown.

**Expected Result**

The test daemonset should error out and give the message shown in the logs, same as the test deployment did.

**Comment by cbron** [comment]
Given latest comments I've transferred back to dashboard. Though IMO this isn't a big blocker for 2.7.2.

**Comment by aalves08** [comment]
@gaktive I will remove the dev-automation label from this issue as it's "impossible" due to the need of cluster provisioning to  automatically test this scenario.

**Comment by richard-cox** [comment]
## On create of the daemonset
Creating a daemonset this way only results in a warning. The UI needs a way to inform the user of this specific warning immediately after the resource is created. 

## The state of a failed daemonset
The state of the Daemonset comes from `metadata.state`. If the state there is incorrect then it is an upstream k8s bug.

We should not massage information from events into the UI's daemonset state given scaling issues. Event count can be huge causing issues to fetch, store and watch for changes. Additionally we are moving away from any kind of table column data that has to be deduced / computed client side, as they break sorting and filtering when using server side pagination.

**Comment by markusewalker** [comment]
Validated that this is addressed on `v2.8-head` https://github.com/rancher/rancher/commit/90fc68836fd8d1bd2996283500f954ace40203ba.

**ENVIRONMENT DETAILS**
- Rancher install: Docker
- Rancher version: `v2.8-head` https://github.com/rancher/rancher/commit/90fc68836fd8d1bd2996283500f954ace40203ba

**TEST RESULT**
|Scenario|Test Case|Result
|---|---|---|
|1|As a standard user, provision a RKE2 cluster with `rancher-restricted` PSACT, deploy a daemonset and validate that a growl appears|:white_check_mark:|

**Scenario 1**
1. As a standard user, provisioned a downstream RKE2 `v1.26.7+rke2r1` node driver cluster.
2. Navigated to Cluster explorer > cluster > Workloads > Daemonsets.
3. Created a daemonset with image as `nginx:latest` in the `default` namespace leads to the expected growl message:
![image](https://github.com/rancher/dashboard/assets/52591819/6ab3d517-1828-44e1-964c-21f7da96c9b4)
